api/server/server.go

```diff
@@ -1137,8 +1137,17 @@ func (s *Server) getContainersByName(version version.Version, w http.ResponseWri
 		return fmt.Errorf("Missing parameter")
 	}
 
+	if err := parseForm(r); err != nil {
+		return err
+	}
+
+	no_err := r.Form.Get("no_err")
 	containerJSON, err := s.daemon.ContainerInspect(vars["name"])
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "no such id") && no_err == "1" {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
I agree with @runcom:
- 404 is correct for 'not found', irrespective of the `no_err` flag. The CLI code you added should treat the 404 as the only reason to move on to look for an image by that name/id. Anything but a 404 is a real error and should stop the processing.
- I don't like the `no_err` flag. The CLI is broken in this space.  Its just so wrong that "docker inspect" looks for both images and containers and https://github.com/docker/docker/pull/13187 is trying to help with that.  But, in the meantime I don't think the API should be modified to add deal with the CLI's poor choices  :-) .  

As was said in https://github.com/docker/docker/issues/10773 I think we should just drop this 'error' down to 'info' or 'debug'.